internal/certification/certificates.go

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2023 F5 Inc. All rights reserved.
+ * Use of this source code is governed by the Apache License that can be found in the LICENSE file.
+ *
+ * Establishes a Watcher for the Kubernetes Secrets that contain the various certificates and keys used to generate a tls.Config object;
+ * exposes the certificates and keys.
+ */
+
+package certification
+
+import (
+	"context"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+// TODO: This needs to use the settings for the secret names...
+
+const (
+	// SecretsNamespace is the value used to filter the Secrets Resource in the Informer.
+	SecretsNamespace = "nlk"
+
+	// CertificateKey is the key for the certificate in the Secret.
+	CertificateKey = "tls.crt"
+
+	// CertificateKeyKey is the key for the certificate key in the Secret.
+	CertificateKeyKey = "tls.key"
+)
+
+type Certificates struct {
+	Certificates map[string]map[string][]byte
+
+	// Context is the context used to control the application.
+	Context context.Context
+
+	// CaCertificateSecretKey is the name of the Secret that contains the Certificate Authority certificate.
+	CaCertificateSecretKey string
+
+	// ClientCertificateSecretKey is the name of the Secret that contains the Client certificate.
+	ClientCertificateSecretKey string
+
+	// informer is the SharedInformer used to watch for changes to the Secrets .
+	informer cache.SharedInformer
+
+	// K8sClient is the Kubernetes client used to communicate with the Kubernetes API.
+	k8sClient kubernetes.Interface
+
+	// eventHandlerRegistration is the object used to track the event handlers with the SharedInformer.
+	eventHandlerRegistration cache.ResourceEventHandlerRegistration
+}
+
+// NewCertificates factory method that returns a new Certificates object.
+func NewCertificates(ctx context.Context, k8sClient kubernetes.Interface) (*Certificates, error) {
+	return &Certificates{
+		k8sClient:    k8sClient,
+		Context:      ctx,
+		Certificates: nil,
+	}, nil
+}
```

Comment:
Why does this return a possible error? Convention? Future proofing?

Reply:
Gone!